docs/src/app/(public)/(content)/react/utils/use-render/page.mdx

```diff
@@ -0,0 +1,78 @@
+# useRender
+
+<Subtitle>Utility for supporting a render prop in custom components.</Subtitle>
+<Meta name="description" content="Utility for supporting a render prop in custom components." />
+
+The `useRender` hook allows you to support the `render` prop to override the default rendered element (similar to `asChild` from Radix) in your custom components.
+
+## API reference
+
+### Input parameters
+
+<PropsReferenceTable
+  data={{
+    render: {
+      type: 'RenderProp<State>',
+      description:
+        'The React element or a function that returns one to override the default element.',
+    },
+    props: {
+      type: 'Record<string, unknown>',
+      description:
+        'Props to be spread on the rendered element.\n\nThey are merged with the internal props\n of the component, so that event handlers are merged,\nclass names and styles are joined, and other external props overwrite the internal ones.',
+    },
+    state: {
```

Comment:
Should we mention that this parameter is optional?

Reply:
I don't think we show optional/required parameter anywhere on the docs yet. We could think of how we want to do this separately. I think it would be better if we just mention the required params/props on the docs instead of the other way around. The type should be enough to guide people.